### ISSUE TITLE
Docker Integration Tests Are Slow and Flaky

### ISSUE THREAD

**Issue by winder** [algorand/indexer#471]
## Summary
The integration tests powered by gnomock/docker are lowering the quality of our testing:

* Each new test adds a fixed 3 second overhead to setup/teardown the docker environment.
* It has become flaky, fairly regularly introducing weird socket errors that appear on random tests and go away after a retry:
```
--- FAIL: TestIgnoreDefaultFrozenConfigUpdate (2.97s)
    postgres_integration_test.go:47: 
        	Error Trace:	postgres_integration_test.go:47
        	            				postgres_integration_test.go:673
        	Error:      	Received unexpected error:
        	            	can't perform last cleanup: read unix @->/run/docker.sock: use of closed network connection
        	Test:       	TestIgnoreDefaultFrozenConfigUpdate
        	Messages:   	Error stoping gnomock
{"level":"info","msg":"Running 13 migrations.","time":"2021-04-28T12:10:56Z"}
```

## Scope/Requirements
Allow injecting a postgres environment in a way that will seamlessly replace gnomock when available.

Environment variables would be an clean way to do this in a way that.

1) Define some variables like `INDEXER_TEST_POSTGRES_URL`, `INDEXER_TEST_POSTGRES_USER`, etc
2) In `postgres_integration_test.go : setupPostgres` check for the environment variables, if available create a database on the provided server. Follow the existing interface, returning sql.DB, connection string, and a cleanup function.
3) In the .travis.yml file, start a postgres instance and set variables before running tests.

## Urgency/Relative Priority
Medium. It's easy to workaround but will continue to become more and more of a problem over time.

**Comment by tolikzinovyev** [comment]
I think we can just remove gnomock. Here is what others do: https://github.com/sfackler/rust-postgres. Also, it would be nice to have the docker image in the repository. It's kind of ridiculous that I have to be connected to the Internet in order to run tests.

**Comment by winder** [comment]
Gnomock makes it trivial to run tests with all dependencies self contained. I'm not sure how relevant an internet connection is relevant to the discussion, the build fetches dependencies from the internet, and most make targets run a `go get` command that requires an internet connection.

My proposal is to support an additional option which would be used the same way as your rust-postgres example

**Comment by tolikzinovyev** [comment]
What dependencies are you referring to?

**Comment by winder** [comment]
go modules

**Comment by tolikzinovyev** [comment]
I'm still not sure what you mean.

**Comment by winder** [comment]
When you build go programs the dependencies are downloaded

**Comment by tolikzinovyev** [comment]
But how does gnomock help?

**Comment by winder** [comment]
It lets you run/debug tests without concerning yourself with any environment setup.

**Comment by tolikzinovyev** [comment]
It basically starts a docker image. We can do the same in the makefile.

**Comment by bricerisingalgorand** [comment]
@chaihoang Did you give this to us on purpose or by mistake?

**Comment by winder** [comment]
Please consult with @tolikzinovyev or me before starting work on this, we have been chipping away at this and are almost complete.
